xExtension-YouTube/extension.php

```diff
@@ -176,29 +190,76 @@ public function getIFrameForLink($link)
     * @param string $link
     * @return string
     */
-    public function getPeerTubeIFrameForLink($link)
+    public function getHtmlPeerTubeContentForLink($entry, $link)
     {
         $url = str_replace('/watch', '/embed', $link);
-        $html = $this->getIFrameHtml($url);
+        $html = $this->getHtml($entry, $url);
         
         return $html;
     }
 
     /**
-     * Returns an HTML <iframe> for a given URL for the configured width and height.
+     * Returns an HTML <iframe> for a given URL for the configured width and height, with content ignored, appended or formated.
      *
      * @param string $url
      * @return string
      */
-    public function getIFrameHtml($url)
+    public function getHtml($entry, $url)
     {
-        return '<iframe class="youtube-plugin-video"
+        $content = '[error]';
+        
+        $iframe = '<iframe class="youtube-plugin-video"
                 style="height: ' . $this->height . 'px; width: ' . $this->width . 'px;" 
                 width="' . $this->width . '" 
                 height="' . $this->height . '" 
                 src="' . $url . '" 
                 frameborder="0" 
-                allowfullscreen></iframe>';
+                allowFullScreen></iframe>';
+        
+        switch ($this->contentHandling) {
+            case 'append':
+                $content = $iframe . $entry->content();
+                break;
+            case 'format':
+                $doc = new DOMDocument();
+                $doc->encoding = 'UTF-8';
+                $doc->recover = true;
+                $doc->strictErrorChecking = false;
+
+                if ($doc->loadHTML('<?xml encoding="utf-8" ?>' . $entry->content()))
+                {
+                    $xpath = new DOMXpath($doc);
+
+                    $titles = $xpath->evaluate("//*[@class='enclosure-title']");
+                    $thumbnails = $xpath->evaluate("//*[@class='enclosure-thumbnail']/@src");
+                    $descriptions = $xpath->evaluate("//*[@class='enclosure-description']");
+
+                    $content = '<div class="enclosure">';
+
+                    if ($titles->length > 0) {
+                        $content .= '<p class="enclosure-title" hidden>' . $titles[0]->nodeValue . '</p>';
```

Comment:
Maybe a comment in the code can help next developers to understand why it's important to add hidden elements? :)

Reply:
Yes ! Done.